io_scene_niftools/kf_export.py

```diff
@@ -0,0 +1,103 @@
+
+# ***** BEGIN LICENSE BLOCK *****
+#
+# Copyright © 2019, NIF File Format Library and Tools contributors.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#    * Redistributions of source code must retain the above copyright
+#      notice, this list of conditions and the following disclaimer.
+#
+#    * Redistributions in binary form must reproduce the above
+#      copyright notice, this list of conditions and the following
+#      disclaimer in the documentation and/or other materials provided
+#      with the distribution.
+#
+#    * Neither the name of the NIF File Format Library and Tools
+#      project nor the names of its contributors may be used to endorse
+#      or promote products derived from this software without specific
+#      prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+# ***** END LICENSE BLOCK *****
+
+import os
+import bpy
+
+import pyffi.spells.nif.fix
+
+from io_scene_niftools.file_io.kf import KFFile
+from io_scene_niftools.modules.nif_export import armature
+from io_scene_niftools.modules.nif_export.animation.transform import TransformAnimation
+from io_scene_niftools.nif_common import NifCommon
+from io_scene_niftools.utils import math
+from io_scene_niftools.utils.singleton import NifOp, NifData
+from io_scene_niftools.utils.logging import NifLog, NifError
+from io_scene_niftools.modules.nif_export import scene
+
+
+class KfExport(NifCommon):
+
+    def __init__(self, operator, context):
+        NifCommon.__init__(self, operator, context)
+
+        # Helper systems
+        self.transform_anim = TransformAnimation()
+
+    def execute(self):
+        """Main import function."""
+
+        NifLog.info(f"Exporting {NifOp.props.filepath}")
+
+        # extract directory, base name, extension
+        directory = os.path.dirname(NifOp.props.filepath)
+        filebase, fileext = os.path.splitext(os.path.basename(NifOp.props.filepath))
+
+        prefix = "x" if bpy.context.scene.niftools_scene.game in ('MORROWIND',) else ""
+        self.version, data = scene.get_version_data()
+        NifData.init(data)
```

Comment:
Store in KFData

Reply:
I would, but this is at times also used to access data version, during the transform functions which are also used by nif export + anim. Very ugly.

Reply:
Can revisit at a later point

Reply:
Yeah I've added a todo. This definitely needs to be addressed.